piet-gpu-derive/src/derive.rs

```diff
@@ -121,6 +121,7 @@ fn gen_derive_ty(ty: &GpuType) -> proc_macro2::TokenStream {
 
 fn gen_derive_scalar_ty(ty: &GpuScalar) -> proc_macro2::TokenStream {
     match ty {
+        GpuScalar::F16 => quote!(crate::f16),
```

Comment:
If you take my "different approach" (see below), then this changes to `half::f16`.